core/inspector.rs

```diff
@@ -220,30 +219,22 @@ impl JsRuntimeInspector {
     loop {
       loop {
         // Do one "handshake" with a newly connected session at a time.
-        if let Some(session) = &mut sessions.handshake {
+        if let Some(mut session) = sessions.handshake.take() {
           let poll_result = session.poll_unpin(cx);
-          let handshake_done =
-            replace(&mut self.flags.borrow_mut().session_handshake_done, false);
           match poll_result {
-            Poll::Pending if handshake_done => {
-              let session = sessions.handshake.take().unwrap();
+            Poll::Pending => {
               sessions.established.push(session);
-              take(&mut self.flags.borrow_mut().waiting_for_session);
+              continue;
             }
-            Poll::Ready(_) => sessions.handshake = None,
-            Poll::Pending => break,
+            Poll::Ready(_) => {}
```

Comment:
Just a suggestion but I think this block would be even easier to understand (and is functionally 100% equivalent) when written like this:
```rust
if let Some(mut session) = sessions.handshake.take() {
  if session.poll_unpin(cx).is_pending() {
    sessions.established.push(session);
  }
}
```
(GH won't let me post it as a diff, unfortunately.)